Source/WebKit/UIProcess/ProvisionalFrameProxy.cpp

```diff
@@ -57,94 +58,67 @@ ProvisionalFrameProxy::ProvisionalFrameProxy(WebFrameProxy& frame, Ref<WebProces
 {
     m_process->markProcessAsRecentlyUsed();
     m_process->addProvisionalFrameProxy(*this);
-
-    m_process->addMessageReceiver(Messages::WebPageProxy::messageReceiverName(), m_pageID, *this);
-
+    // Its possible that this is called with a WebFrameProxy which has the message already registered (SiteIsolation.IframeNavigatesSelfWithoutChangingOrigin)
+    if (frame.processIdentifier() == m_process->processIdentifier()) {
```

Comment:
This check should be removed.  The ProvisionalFrameProxy constructor probably shouldn't have a WebFrameProxy or a request.

Reply:
That should allow you to also remove bool newProcess.

Reply:
It's not clear how this condition can be removed -- this is needed because `SiteIsolation.IframeNavigatesSelfWithoutChangingOrigin` runs into the case where the same WebFrameProxy is used to create a new ProvisionalFrameProxy. If we remove this, what will register for these messages from the web process? We could have SubframePageProxy do that, but that won't have a 1-1 mapping.